core/src/trezor/crypto/rlp.py

```diff
@@ -1,53 +1,94 @@
-def int_to_bytes(x: int) -> bytes:
-    if x == 0:
-        return b""
-    r = bytearray()
-    while x:
-        r.append(x % 256)
-        x //= 256
-    return bytes(reversed(r))
-
-
-def encode_length(l: int, is_list: bool) -> bytes:
-    offset = 0xC0 if is_list else 0x80
-    if l < 56:
-        return bytes([l + offset])
-    elif l < 256 ** 8:
-        bl = int_to_bytes(l)
-        return bytes([len(bl) + offset + 55]) + bl
+from micropython import const
+
+if False:
+    from typing import Union
+    from trezor.utils import Writer
+
+    RLPItem = Union["RLPItem", bytes, int]
```

Comment:
What does the recursion do here?

Also, should there be bytearray since it's checked against in the code?

Reply:
Ah, I meant to run this though mypy and forgot. Good catch, this is supposed to be something like `Union[list["RLPItem"], bytes, int]`

(also I suspect that mypy won't actually allow a recursive type)

Reply:
4a009c8